tensorflow_datasets/image/vis_dial.py

```diff
@@ -0,0 +1,137 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Visual Dialog Dataset"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+import re
+import fnmatch
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """\
+@inproceedings{visdial,
+  title={{V}isual {D}ialog},
+  author={Abhishek Das and Satwik Kottur and Khushi Gupta and Avi Singh
+    and Deshraj Yadav and Jos\'e M.F. Moura and Devi Parikh and Dhruv Batra},
+  booktitle={Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition},
+  year={2017}
+}
+"""
+
+_TRAIN_DIALOGS = "https://dl.dropboxusercontent.com/s/ix8keeudqrd8hn8/visdial_1.0_train.zip"
+_TRAIN_IMAGES = "http://images.cocodataset.org/zips/train2014.zip"
+_VALIDATION_DIALOGS = "https://dl.dropboxusercontent.com/s/ibs3a0zhw74zisc/visdial_1.0_val.zip"
+_VALIDATION_IMAGES = "https://dl.dropboxusercontent.com/s/twmtutniktom7tu/VisualDialog_val2018.zip"
+_TEST_DIALOGS = "https://dl.dropboxusercontent.com/s/o7mucbre2zm7i5n/visdial_1.0_test.zip"
+_TEST_IMAGES = "https://dl.dropboxusercontent.com/s/mwlrg31hx0430mt/VisualDialog_test2018.zip"
+
+
+class VisualDialog(tfds.core.GeneratorBasedBuilder):
+    """Visual Dialog Dataset"""
+
+    VERSION = tfds.core.Version("1.0.0")
+
+    def _info(self):
```

Comment:
Please check this [link](https://www.tensorflow.org/community/contribute/code_style) about tensorflow code style.